docs/Technical Details/index.rst

```diff
@@ -0,0 +1,189 @@
+*****************
+Technical Details
+*****************
+
+easy_infra.yml
+==============
+
+``easy_infra`` is a project which makes heavy use of generated code and composable container images. ``easy_infra.yml`` is the centralized
+configuration to instruct those generation and composition processes. It describes which version of software to use where, how to generate the very
+important ``/functions`` script (more on that in `/functions and functions.j2`_), which is what provides all of the hooking and capabilities.
+
+Here is a ficticious ``easy_infra.yml`` that concisely demonstrates the various features that are possible::
+
+    _anchors:
+      file_extensions: &id001
+      - tf
+      security: &id002
+        checkov:
+          command: checkov -d . --download-external-modules True --skip-download --output
+            json --output-file-path ${CHECKOV_JSON_REPORT_PATH}/checkov.json,
+          customizations:
+            CHECKOV_BASELINE: --baseline
+          description: directory scan
+      validation: &id003
+      - command: terraform init -backend=false
+        description: initialization
+      - command: terraform validate
+        description: validation
+    commands:
+      checkov:
+        allow_update: False
+        version: 2.2.8
+        version_argument: --version
+      tfenv:
+        aliases:
+        - tfe
+        allow_filter:
+        - match: exec
+          position: 0
+        file_extensions: *id001
+        security: *id002
+        validation: *id003
+        version: v3.0.0
+        version_argument: --version
+
+.. note::
+    Not all of these features are used in the default ``easy_infra`` images
+
+YML Anchors and Aliases
+-----------------------
+
+Anchors and Aliases are a yml concept, and are fully supported in ``easy_infra.yml`` to support reusable components. See ``&id001`` and ``*id001`` above to see
+an example. Learn more `here <https://yaml.org/spec/1.2.2/#3222-anchors-and-aliases>`_ and `here
+<https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/>`_.
+
+Commands
+--------
+
+All of the terms under ``commands`` are the names of packages or tools, and the details of how and when they are installed are described in that
+object of the ``easy_infra.yml``. In order to register a runtime hook against a command (Learn more about hooks `here <../Hooks/index.html>`_), or define
```

Comment:
Link does not work

Reply:
Hmm it's working for me?  I fixed this in [2b79c36](https://github.com/SeisoLLC/easy_infra/pull/221/commits/2b79c368444c6a7992ed10bad9130137a0154ca8) 🤔

Reply:
Strange, its resolving to this link for me and it is still not working: https://github.com/SeisoLLC/easy_infra/blob/remove-validation-add-docs/docs/Hooks/index.html
<img width="1435" alt="Screen Shot 2022-11-09 at 8 41 44 AM" src="https://user-images.githubusercontent.com/97695696/200845411-cee45212-9175-4767-9ba8-7aa0170f7486.png">

Reply:
@digeseiso See the "Testing" section of the PR.  The `docs/` folder is for read the docs, not github.

Reply:
@JonZeolla ahh make sense now